### PR TITLE
Qt: Work around broken drivers that don't allow backend switching

### PR DESCRIPTION
**Pull request by spycrab**
Praying to the Qt gods this doesn't break literally everything.

### CODE REVIEW

**Comment by lioncash** [comment]
There should probably be a comment somewhere explaining that the code works around issues, so someone else doesn't inadvertently break it if changes are made in the future.

**Comment by spycrab** [comment]
@lioncash Done.

File: Source/Core/DolphinQt2/MainWindow.cpp [hunk 1]
```diff
@@ -575,7 +575,21 @@ void MainWindow::HideRenderWidget()
     disconnect(Host::GetInstance(), &Host::RequestTitle, this, &MainWindow::setWindowTitle);
     setWindowTitle(QString::fromStdString(Common::scm_rev_str));
   }
+
+  // The following code works around a driver bug that would lead to Dolphin crashing when changing
+  // graphics backends (e.g. OpenGL to Vulkan) to avoid this the render widget is (safely) recreated
```
Comment (leoetlino) [inline Source/Core/DolphinQt2/MainWindow.cpp]: Nit: it feels like "to avoid this..." should be its own sentence.

Comment (spycrab) [reply Source/Core/DolphinQt2/MainWindow.cpp]: Done.

**Pull request merged**